core/src/main/scala/org/apache/spark/util/ListenerBus.scala

```diff
@@ -37,6 +38,12 @@ private[spark] trait ListenerBus[L <: AnyRef, E] extends Logging {
   // Marked `private[spark]` for access in tests.
   private[spark] def listeners = listenersPlusTimers.asScala.map(_._1).asJava
 
+  private lazy val logSlowEventEnabled =
```

Comment:
SparkEnv may not be always available, especially for test suites. The test failures in CI builds comes from NPE, as NPE occurs here, and NPE gets propagated and finally kills the dispatch thread for AsyncEventQueue.

So we might want to set default value if SparkEnv is not available, or have some method for descendants to provide SparkConf.